x-pack/plugin/ilm/qa/multi-cluster/src/test/java/org/elasticsearch/xpack/indexlifecycle/CCRIndexLifecycleIT.java

```diff
@@ -0,0 +1,287 @@
+                    "\"aliases\": {\"" + alias + "\":  {\"is_write_index\":  true}} }");
+                assertOK(leaderClient.performRequest(createIndexRequest));
+                // Check that the new index is creeg
+                Request checkIndexRequest = new Request("GET", "/_cluster/health/" + indexName);
+                checkIndexRequest.addParameter("wait_for_status", "green");
+                checkIndexRequest.addParameter("timeout", "70s");
+                checkIndexRequest.addParameter("level", "shards");
+                assertOK(leaderClient.performRequest(checkIndexRequest));
+
+                // Check that it got replicated to the follower
+                assertBusy(() -> assertTrue(indexExists(indexName)));
+
+                // Aliases are not copied from leader index, so we need to add that for the rollover action in follower cluster:
+                client().performRequest(new Request("PUT", "/" + indexName + "/_alias/" + alias));
+
+                index(leaderClient, indexName, "1");
+                assertDocumentExists(leaderClient, indexName, "1");
+
+                assertBusy(() -> {
+                    assertDocumentExists(client(), indexName, "1");
+                    // Sanity check that following_index setting has been set, so that we can verify later that this setting has been unset:
+                    assertThat(getIndexSetting(client(), indexName, "index.xpack.ccr.following_index"), equalTo("true"));
+                });
+
+                // Wait for the index to roll over on the leader
+                assertBusy(() -> {
+                    assertOK(leaderClient.performRequest(new Request("HEAD", "/" + nextIndexName)));
+                    assertThat(getIndexSetting(leaderClient, indexName, "index.lifecycle.indexing_complete"), equalTo("true"));
+
+                });
+
+                assertBusy(() -> {
+                    // Wait for the next index should have been created on the leader
+                    assertOK(leaderClient.performRequest(new Request("HEAD", "/" + nextIndexName)));
+                    // And the old index should have a write block and indexing complete set
+                    assertThat(getIndexSetting(leaderClient, indexName, "index.blocks.write"), equalTo("true"));
+                    assertThat(getIndexSetting(leaderClient, indexName, "index.lifecycle.indexing_complete"), equalTo("true"));
+
+                });
+
+                assertBusy(() -> {
+                    // Wait for the setting to get replicated to the follower
+                    assertThat(getIndexSetting(client(), indexName, "index.lifecycle.indexing_complete"), equalTo("true"));
+                }, 2, TimeUnit.MINUTES);
+                // TODO: ^^ I don't know why but this takes like 70 seconds to happen. It doesn't seem like it should and this should
+                // definitely be fixed before merging.
```

Comment:
I don't know what's up here, @martijnvg can we take a look at this when you're back?

Reply:
I pushed to following fix: 394944510484d31521f51333d3f31f5dc3fde650

The ccr read poll interval needed to be decreased.

The shard follow tasks continuesly poll changes from leader index,
in case no new writes arrive in the leader shards, then the leader
shards wait up to the defined read_poll_timeout before returning an
empty result. The result always includes meta information about whether
the index settings need to be synced.

Because in this test no data was indexed it took a full minute for ccr
to realize that the follower index settings needed to be updated and
caused the test to fail due to an assert busy statement to time out.

Reply:
Thanks! Figured it was something like that, glad it was a simple fix.